src/base/api.lua

```diff
@@ -1092,6 +1092,31 @@
 	})
 
 
+--
+-- nested data kind; wraps simple values into a table, returns others as-is.
+--
+
+	premake.field.kind("nested", {
+		store = function(field, current, value, processor)
+			-- set each value.
+			local result = {}
+			for name, v in pairs(value) do
+				local f = field.fields[name]
```

Comment:
No... field.fields is the table inside the field, it's not touching any global tables.